### ISSUE TITLE
Default user manual pdf 400 bad request from a mangled webdav URL in 15 (docker)

### ISSUE THREAD

**Issue by mcarpenterjr** [nextcloud/server#13669]
### Steps to reproduce
1.Spin up a new docker instance
2.Login
3. attempt to download/open the PDF user manual

### Expected behaviour
User manual should open in the pdf module if enabled or fallback to browser default

### Actual behaviour
Returns 400 bad request.

### Server configuration

**Operating system**:

**Web server:**

**Database:**

**PHP version:**
7.3
**Nextcloud version:** (see Nextcloud admin page)
15.0.2
**Updated from an older Nextcloud/ownCloud or fresh install:**
Fresh Docker Container
**Where did you install Nextcloud from:**
Docker Registry
**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - accessibility: 1.1.0
  - bruteforcesettings: 1.3.0
  - calendar: 1.6.4
  - cloud_federation_api: 0.1.0
  - comments: 1.5.0
  - contacts: 3.0.1
  - dav: 1.8.1
  - deck: 0.5.2
  - federatedfilesharing: 1.5.0
  - federation: 1.5.0
  - files: 1.10.0
  - files_pdfviewer: 1.4.0
  - files_sharing: 1.7.0
  - files_texteditor: 2.7.0
  - files_trashbin: 1.5.0
  - files_versions: 1.8.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.3.0
  - notifications: 2.3.0
  - oauth2: 1.3.0
  - password_policy: 1.5.0
  - provisioning_api: 1.5.0
  - serverinfo: 1.5.0
  - sharebymail: 1.5.0
  - support: 1.0.0
  - theming: 1.6.0
  - twofactor_backupcodes: 1.4.1
  - updatenotification: 1.5.0
  - user_ldap: 1.5.0
  - workflowengine: 1.5.0
Disabled:
  - activity
  - admin_audit
  - encryption
  - files_external
  - files_videoplayer
  - firstrunwizard
  - gallery
  - nextcloud_announcements
  - survey_client
  - systemtags
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "htaccess.RewriteBase": "\/",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "10.27.1.130:8380",
            "calendar"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "15.0.2.0",
        "overwrite.cli.url": "http:\/\/10.27.1.130:8380",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "ldapIgnoreNamingRules": false,
        "ldapProviderFactory": "OCA\\User_LDAP\\LDAPProviderFactory"
    },
    "apps": {
        "accessibility": {
            "enabled": "yes",
            "installed_version": "1.1.0",
            "types": ""
        },
        "activity": {
            "enabled": "no",
            "installed_version": "2.8.2",
            "types": "filesystem"
        },
        "backgroundjob": {
            "lastjob": "10"
        },
        "bruteforcesettings": {
            "enabled": "yes",
            "installed_version": "1.3.0",
            "types": ""
        },
        "calendar": {
            "enabled": "yes",
            "installed_version": "1.6.4",
            "types": ""
        },
        "cloud_federation_api": {
            "enabled": "yes",
            "installed_version": "0.1.0",
            "types": "filesystem"
        },
        "comments": {
            "enabled": "yes",
            "installed_version": "1.5.0",
            "types": "logging"
        },
        "contacts": {
            "enabled": "yes",
            "installed_version": "3.0.1",
            "types": ""
        },
        "core": {
            "installed.bundles": "[\"CoreBundle\"]",
            "installedat": "1547749121.5",
            "lastcron": "1547834640",
            "lastupdateResult": "[]",
            "lastupdatedat": "1547821952",
            "oc.integritycheck.checker": "[]",
            "public_files": "files_sharing\/public.php",
            "public_webdav": "dav\/appinfo\/v1\/publicwebdav.php",
            "scss.variables": "746d374ec1ad891bbce3d4bb92b5e517",
            "vendor": "nextcloud"
        },
        "dav": {
            "enabled": "yes",
            "installed_version": "1.8.1",
            "types": "filesystem"
        },
        "deck": {
            "enabled": "yes",
            "installed_version": "0.5.2",
            "types": "logging"
        },
        "federatedfilesharing": {
            "enabled": "yes",
            "installed_version": "1.5.0",
            "types": ""
        },
        "federation": {
            "enabled": "yes",
            "installed_version": "1.5.0",
            "types": "authentication"
        },
        "files": {
            "cronjob_scan_files": "500",
            "enabled": "yes",
            "installed_version": "1.10.0",
            "types": "filesystem"
        },
        "files_pdfviewer": {
            "enabled": "yes",
            "installed_version": "1.4.0",
            "types": ""
        },
        "files_sharing": {
            "enabled": "yes",
            "installed_version": "1.7.0",
            "types": "filesystem"
        },
        "files_texteditor": {
            "enabled": "yes",
            "installed_version": "2.7.0",
            "types": ""
        },
        "files_trashbin": {
            "enabled": "yes",
            "installed_version": "1.5.0",
            "types": "filesystem,dav"
        },
        "files_versions": {
            "enabled": "yes",
            "installed_version": "1.8.0",
            "types": "filesystem,dav"
        },
        "files_videoplayer": {
            "enabled": "no",
            "installed_version": "1.4.0",
            "types": ""
        },
        "firstrunwizard": {
            "enabled": "no",
            "installed_version": "2.4.0",
            "types": "logging"
        },
        "gallery": {
            "enabled": "no",
            "installed_version": "18.2.0",
            "types": ""
        },
        "logreader": {
            "enabled": "yes",
            "installed_version": "2.0.0",
            "types": ""
        },
        "lookup_server_connector": {
            "enabled": "yes",
            "installed_version": "1.3.0",
            "types": "authentication"
        },
        "nextcloud_announcements": {
            "enabled": "no",
            "installed_version": "1.4.0",
            "types": "logging"
        },
        "notifications": {
            "enabled": "yes",
            "installed_version": "2.3.0",
            "types": "logging"
        },
        "oauth2": {
            "enabled": "yes",
            "installed_version": "1.3.0",
            "types": "authentication"
        },
        "password_policy": {
            "enabled": "yes",
            "installed_version": "1.5.0",
            "types": ""
        },
        "provisioning_api": {
            "enabled": "yes",
            "installed_version": "1.5.0",
            "types": "prevent_group_restriction"
        },
        "serverinfo": {
            "enabled": "yes",
            "installed_version": "1.5.0",
            "types": ""
        },
        "sharebymail": {
            "enabled": "yes",
            "installed_version": "1.5.0",
            "types": "filesystem"
        },
        "support": {
            "enabled": "yes",
            "installed_version": "1.0.0",
            "types": ""
        },
        "survey_client": {
            "enabled": "no",
            "installed_version": "1.3.0",
            "types": ""
        },
        "systemtags": {
            "enabled": "no",
            "installed_version": "1.5.0",
            "types": "logging"
        },
        "theming": {
            "backgroundMime": "backgroundColor",
            "cachebuster": "9",
            "color": "#0D4524",
            "enabled": "yes",
            "faviconMime": "image\/png",
            "installed_version": "1.6.0",
            "logoMime": "image\/png",
            "logoheaderMime": "image\/png",
            "name": "SMT Calendar",
            "slogan": "Personnel and Event scheduling",
            "types": "logging",
            "url": "http:\/\/calendar\/"
        },
        "twofactor_backupcodes": {
            "enabled": "yes",
            "installed_version": "1.4.1",
            "types": ""
        },
        "updatenotification": {
            "enabled": "yes",
            "installed_version": "1.5.0",
            "types": "",
            "update_check_errors": "0"
        },
        "user_ldap": {
            "cleanUpJobOffset": "0",
            "enabled": "yes",
            "installed_version": "1.5.0",
            "s01_lastChange": "1547825243",
            "s01has_memberof_filter_support": "1",
            "s01home_folder_naming_rule": "",
            "s01last_jpegPhoto_lookup": "0",
            "s01ldap_agent_password": "bGRhcHVzZXI=",
            "s01ldap_attributes_for_group_search": "",
            "s01ldap_attributes_for_user_search": "",
            "s01ldap_backup_host": "",
            "s01ldap_backup_port": "",
            "s01ldap_base": "cn=users,dc=internal,dc=smtcorp,dc=com",
            "s01ldap_base_groups": "cn=users,dc=internal,dc=smtcorp,dc=com",
            "s01ldap_base_users": "cn=users,dc=internal,dc=smtcorp,dc=com",
            "s01ldap_cache_ttl": "600",
            "s01ldap_configuration_active": "1",
            "s01ldap_default_ppolicy_dn": "",
            "s01ldap_display_name": "displayname",
            "s01ldap_dn": "ldap@internal.smtcorp.com",
            "s01ldap_dynamic_group_member_url": "",
            "s01ldap_email_attr": "mail",
            "s01ldap_experienced_admin": "0",
            "s01ldap_expert_username_attr": "samaccountname",
            "s01ldap_expert_uuid_group_attr": "",
            "s01ldap_expert_uuid_user_attr": "",
            "s01ldap_gid_number": "gidNumber",
            "s01ldap_group_display_name": "cn",
            "s01ldap_group_filter": "",
            "s01ldap_group_filter_mode": "0",
            "s01ldap_group_member_assoc_attribute": "uniqueMember",
            "s01ldap_groupfilter_groups": "",
            "s01ldap_groupfilter_objectclass": "",
            "s01ldap_host": "winserver",
            "s01ldap_login_filter": "(&(&(|(objectclass=person)))(samaccountname=%uid))",
            "s01ldap_login_filter_mode": "0",
            "s01ldap_loginfilter_attributes": "",
            "s01ldap_loginfilter_email": "0",
            "s01ldap_loginfilter_username": "1",
            "s01ldap_nested_groups": "0",
            "s01ldap_override_main_server": "",
            "s01ldap_paging_size": "500",
            "s01ldap_port": "389",
            "s01ldap_quota_attr": "",
            "s01ldap_quota_def": "",
            "s01ldap_tls": "0",
            "s01ldap_turn_off_cert_check": "0",
            "s01ldap_turn_on_pwd_change": "0",
            "s01ldap_user_avatar_rule": "default",
            "s01ldap_user_display_name_2": "",
            "s01ldap_user_filter_mode": "1",
            "s01ldap_userfilter_groups": "",
            "s01ldap_userfilter_objectclass": "person",
            "s01ldap_userlist_filter": "(|(&(objectCategory=group)(mail=*))(&(objectCategory=person)(objectClass=user)(mail=*)(!(userAccountControl:1.2.840.113556.1.4.803:=2))))",
            "s01use_memberof_to_detect_membership": "1",
            "types": "authentication"
        },
        "workflowengine": {
            "enabled": "yes",
            "installed_version": "1.5.0",
            "types": "filesystem"
        }
    }
}

```
</details>

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/11430497/51406339-dfdcaf80-1b26-11e9-86f5-7625944d3f14.gif)

**Comment by mcarpenterjr** [comment]
Right off the bat, I noticed the URL doesn't match when accessing another dav item in next cloud. In the interim these files don't matter in this case as we're only using the calendar aspect of next cloud in this instance and we'll most likely turn off the directory skeleton and reprovision the users.

**Comment by skjnldsv** [comment]
@MorrisJobke

**Comment by MorrisJobke** [comment]
Fixed in the recent releases (15.0.4 for example)
